pkg/cli/alpha/config-gen/cmd.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configgen
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/markbates/pkger"
+	"github.com/spf13/cobra"
+
+	// import pkged files
+	_ "sigs.k8s.io/kubebuilder/v3"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// NewCommand returns a new cobra command
+func NewCommand() *cobra.Command {
+	kp := &KubebuilderConfigGen{}
+
+	// legacy kustomize function support
+	legacyPlugin := os.Getenv("KUSTOMIZE_PLUGIN_CONFIG_STRING")
+	err := yaml.Unmarshal([]byte(legacyPlugin), kp)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		return nil
+	}
+
+	// Eager check to make sure pkged templates are found.
+	err = pkger.Walk("/pkg/cli/alpha/config-gen/templates/resources", func(_ string, _ os.FileInfo, err error) error {
+		return err
+	})
+	if err != nil {
+		// this shouldn't fail it it was compiled correctly
+		fmt.Fprintln(os.Stderr, err.Error())
+		return nil
+	}
+
+	c := framework.TemplateCommand{
+		API: kp,
+
+		MergeResources: true, // apply additional inputs as patches
+
+		// these are run before the templates
+		PreProcessFilters: []kio.Filter{
+			// run controller-gen libraries to generate configuration from code
+			ControllerGenFilter{KubebuilderConfigGen: kp},
+			// inject generated certificates
+			CertFilter{KubebuilderConfigGen: kp},
+		},
+
+		// generate resources
+		TemplatesFn: framework.TemplatesFromDir(pkger.Dir("/pkg/cli/alpha/config-gen/templates/resources")),
```

Comment:
NIT: redundant type conversion:
```suggestion
		TemplatesFn: framework.TemplatesFromDir("/pkg/cli/alpha/config-gen/templates/resources"),
```